chia/simulator/full_node_simulator.py

```diff
@@ -144,3 +203,288 @@ async def reorg_from_index_to_new_index(self, request: ReorgProtocol):
+            The total number of reward mojos farmed to the requested address.
+        """
+        if isinstance(timeout, _Default):
+            timeout = (count + 1) * timeout_per_block
+            timeout += 5
+
+        with anyio.fail_after(delay=timeout):
+            if count == 0:
+                return 0
+
+            target_puzzlehash = await wallet.get_new_puzzlehash()
+            rewards = 0
+
+            block_reward_coins = set()
+            expected_reward_coin_count = 2 * count
+
+            # TODO: why two final transaction blocks and not just one?
+            for to_wallet in [*([True] * count), False, False]:
+                if to_wallet:
+                    rewards += await self.farm_blocks_to_puzzlehash(
+                        count=1,
+                        farm_to=target_puzzlehash,
+                        guarantee_transaction_blocks=False,
+                        timeout=None,
+                    )
+                else:
+                    await self.farm_blocks_to_puzzlehash(count=1, guarantee_transaction_blocks=True, timeout=None)
+
+                peak_height = self.full_node.blockchain.get_peak_height()
+                if peak_height is None:
+                    raise RuntimeError("Peak height still None after processing at least one block")
+
+                coin_records = await self.full_node.coin_store.get_coins_added_at_height(height=peak_height)
+                for record in coin_records:
+                    if record.coin.puzzle_hash == target_puzzlehash and record.coinbase:
+                        block_reward_coins.add(record.coin)
+
+                if len(block_reward_coins) >= expected_reward_coin_count:
+                    break
+            else:
+                raise RuntimeError("Not all reward coins identified")
+
+            if len(block_reward_coins) != expected_reward_coin_count:
+                raise RuntimeError(
+                    f"Expected {expected_reward_coin_count} reward coins, got: {len(block_reward_coins)}"
+                )
+
+            await wait_for_coins_in_wallet(coins=block_reward_coins, wallet=wallet, timeout=None)
+
+            return rewards
+
+    async def farm_rewards_to_wallet(
+        self,
+        amount: int,
+        wallet: Wallet,
+        timeout: Union[None, _Default, float] = default,
+    ) -> int:
+        """Farm at least the requested amount of mojos to the passed wallet. Extra
+        mojos will be received based on the block rewards at the present block height.
+        The rewards will be present in the wall before returning.
+
+        Arguments:
+            amount: The minimum number of mojos to farm.
+            wallet: The wallet to farm the block rewards to.
+
+        Returns:
+            The total number of reward mojos farmed to the requested wallet.
+        """
+        rewards = 0
+
+        if amount == 0:
+            return rewards
+
+        height_before = self.full_node.blockchain.get_peak_height()
```

Comment:
can we use type annotations? For example: `Optional[uint32]`

Reply:
I'll add it but I dislike this since it can broaden the hinting unnecessarily which can cause extra handling code to be introduced downstream.  I've added this to a list to discuss later in terms of preferred coding practices.